### PR TITLE
Feature/background component

### PR DESCRIPTION
**Pull request by PriyaRajarathinam**
Problem
=======
As an engineer, I want a baseline background for cross-sectional components.

[Link to Asana Ticket](https://app.asana.com/0/1197738625416321/1199332791703155)

Solution
========

I implemented a Background component that creates an overarching background for the sections. There is a separate background for Sections 1-4 and a separate background for Sections 5-7. Sections 1-4 have the basic background with a blue wire frame and Sections 5-7 have a gradient background with a rainbow wire frame.

Change summary:
---------------
* Added background component and imported new images
* added background to the home component

Steps to Verify:
----------------
1. Start the webserver to see the feature
2. It looks nice with only seven sections (4 in section 1 and 3 in section 2)

Screenshots (optional):
-----------------------
Show-n-tell images/animations here

### CODE REVIEW

File: src/components/Background/index.view.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from "react";
```
Comment (ilopezro) [inline src/components/Background/index.view.tsx:1]: You should take a look at the changes introduced in PR #71 when upgrading `typescript` to 4.1+. We don't need to import react anymore unless we need it. Typescript, in general, we import react the following way: 
```suggestion
import * as React from "react";
```

File: src/components/Background/index.view.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+const Background: React.FC<BackgroundProps> = ({
+  children,
+}: BackgroundProps) => {
+  return (
+    <div>
+      <div className="Background">
+        <div className="Background__children">
+          {Array.isArray(children)
+            ? children.filter((child, index) => index < 4)
+            : children}
```
Comment (ilopezro) [inline src/components/Background/index.view.tsx:18]: There's no need to check to see if `children` prop is an array. We can simply just drop the `children` props in here

File: src/images/wire.svg [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+<defs>
+<linearGradient id="paint0_linear" x1="1017.01" y1="0.138689" x2="357.536" y2="3536.37" gradientUnits="userSpaceOnUse">
+<stop offset="0.0165518" stop-color="#4366C0" stop-opacity="0"/>
+<stop offset="0.10046" stop-color="#4366C0" stop-opacity="0.3"/>
+<stop offset="0.244792" stop-color="#4366C0" stop-opacity="0.3"/>
+<stop offset="0.837272" stop-color="#4366C0" stop-opacity="0.3"/>
+<stop offset="0.963542" stop-color="#4366C0"/>
+</linearGradient>
+</defs>
+</svg>
```
Comment (ilopezro) [inline src/images/wire.svg:12]: Super nit picky, but could we add an end-line here?

File: src/stylesheets/_variables.scss [hunk 4]
```diff
@@ -4,13 +4,16 @@
...
+  --darkmode: #1a1542;
+  --blue: #4366c0;
 
   /* variations */
   --wedgewood__dark: #{darken(#4f728e, 7%)};
 
   /* These are for gradient */
   --wedgewood__gradient: #486988;
   --waterleaf: #a9ebdf;
+  --blue_gradient: #344790;
```
Comment (ilopezro) [inline src/stylesheets/_variables.scss:16]: In order to be more in line with the BEM naming conventions, we should have this be `blue__gradient`(with the double underscore) or `blueGradient`

File: src/views/Hero/Hero.scss [hunk 5]
```diff
@@ -7,6 +7,7 @@
   min-height: 100%;
 
   &__homeView {
+    /*
```
Comment (ilopezro) [inline src/views/Hero/Hero.scss:10]: I'm wondering if we could just delete all these comments. If we no longer need these lines of code, let's just get rid of them!

File: src/components/Background/index.view.tsx (hunk 2 above)
Comment (PriyaRajarathinam) [reply src/components/Background/index.view.tsx:18]: If I simply drop the children props how would I section the pages?

Comment (ilopezro) [reply src/components/Background/index.view.tsx:18]: I think "drop" would be the wrong word here. What I was trying to say was that we can simply have this block be: 
```suggestion
          {children}
```

File: src/components/Background/index.view.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+const Background: React.FC<BackgroundProps> = ({
+  children,
+}: BackgroundProps) => {
+  return (
+    <div className="Background">
+      <div className="Background__gradient"></div>
+      <div className="Background__children">{children}</div>
+      <Wire className="Background__wire"></Wire>
+      <Rainbow className="Background__wire--variant"></Rainbow>
+    </div>
```
Comment (ilopezro) [inline src/components/Background/index.view.tsx:18]: 🎉

**Review by ilopezro — COMMENTED** [review]
This looks really good to me! I just have a super quick question: is this the intended view right now? I see that when I resize I can see the rest of the background. My guess is that as we keep adding components, the background will slowly appear where it's supposed to be

Screen Shot 2020-12-03 at 12 56 22 AM

File: src/components/Background/index.view.tsx (hunk 6 above)
Comment (PriyaRajarathinam) [reply src/components/Background/index.view.tsx:18]: Yup! It's meant to work with 7 sections.

Comment (w-gao) [reply src/components/Background/index.view.tsx:18]: This looks amazing with more sections added in!

**Review by mocklee — COMMENTED** [review]
cool progress so far! let's flush out further details with Calum tomorrow during the meeting

File: src/components/Background/Background.scss [hunk 7]
```diff
@@ -0,0 +1,40 @@
+@use "~stylesheets/typography";
+
+.Background {
```
[the remark this answers is not in the document]
Comment (tim-nguyen-cs) [reply src/components/Background/Background.scss:3]: Could we even do `7000px` for each of the supposed 7 sections?

File: src/components/Background/Background.scss [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    z-index: -2;
+    position: absolute;
+    background: linear-gradient(
+      180deg,
+      var(--blue) 0%,
+      var(--blue__gradient) 100%
+    );
+    width: 100%;
+    height: calc(100% / 7 * 3);
+    bottom: 0;
```
Comment (tim-nguyen-cs) [inline src/components/Background/Background.scss:38]: Do you think we could rearrange the SCSS to replicate the structure of the HTML? It'll just make things easier to understand.

File: src/components/Background/Background.scss [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+    &--variant {
+      position: absolute;
+      width: 100%;
+      bottom: 0;
+    }
+  }
+
+  &__children {
+    z-index: 999;
```
Comment (tim-nguyen-cs) [inline src/components/Background/Background.scss:24]: Can we also change to to something more realistic? Like 0 or 1? My components have z-indices too, but I'm not sure how that might clash.

**Review by tim-nguyen-cs — COMMENTED** [review]
Everything looks good to me! I love the absolute/relative positioning magin - great job @PriyaRajarathinam !

File: src/components/Background/Background.scss (hunk 7 above)
[the remark this answers is not in the document]
Comment (mocklee) [reply src/components/Background/Background.scss:3]: hopefully we won't need to define a height if the sections size up well against the background

**Review by mocklee — APPROVED** [review]
I think the background is ready for staging! let's get this into `main` & go from there..

**Pull request merged**